keps/sig-node/2837-pod-level-resource-spec/README.md

```diff
@@ -0,0 +1,1108 @@
+
+* **BestEffort QoS Criteria**
+
+    None of pod-level or container-level resources requests or limits are specified.
+
+#### Init Containers
+
+With the current implementation (before this KEP), Pod’s effective resource request/limit is the higher of:
+
+* The sum of all app containers request/limit for a resource.
+* The highest value resource request or limit among all the init containers.
+
+With the support of pod-level requests and limits, following are a few options that can be used to determine pod’s effective resource request/limit:
+
+* **Option 1: Static values of effective pod request/limit as supplied in the PodSpec.**
+
+  In this the pod level cgroup will be set using the requests and limits from the resources specified at pod-level.
+
+  Example pod spec:
+
+  ```yaml
+  apiVersion: v1
+  kind: Pod
+  spec:
+    resources:
+      requests:
+        memory: 100Mi
+        cpu: 1
+      limits:
+        memory: 200Mi
+        cpu: 2
+    containers:
+      - name: container-1
+      - name: container-2
+        resources:
+          requests:
+            memory: 50Mi
+            cpu: 1
+  ```
+
+  **Resources requests at pod-level**
+
+  * The scheduler will use the requested resources (CPU and memory) settings specified in the PodSpec to find a suitable node to place the pod. 
+  * Kubelet will use the CPU requests from the pod-level resources section, and translate it to cpu.weight in pod-level cgroup.
+
+  **Resources limits at pod-level**
+
+  * All init and non-init containers will be capped by the pod level limit which represents the peak usage of all containers.
+  * Kubelet will use CPU and memory limits from PodSpec, and translate them to cpu.max and memory.max cgroup settings at pod-level respectively. 
+
+  **Pros**
+
+  * Simple to understand and implement.
+  * Guarantees enough resources for all containers.
+
+  **Cons**
+
+  * Can lead to over-provisioning (resource wastage), especially if the init container peak usage is significantly higher than the steady-state usage 
+  of the other containers. 
+
+  If this option is picked, it might be worth providing guidance and examples in the feature documentation to help users effectively set pod-level requests and limits. 
+  
+  Example 1: If your application containers require high resources simultaneously, it might be worth setting pod-level requests/limits such that it is an aggregate of all
+  container-level requests/limits.
+
+  Example 2: If your application has init containers with high resource requirements and usage, while your application containers require fairly less resources, it might be
+  worth setting pod-level requests/limits as maximum of init and non-init container resources requirements. 
+
+  Since init containers run sequentially, there is no requirement to share resources with any other container. Therefore setting resource requests and limits as the higher value 
```

Comment:
except for the sidecars

Reply:
expanding on this: sidecar containers will start sequentially and then run while next init containers are running. So resources usage calculation should account for sidecar container + max of init containers running after this sidecar container